cmake/package.pc.in

```diff
@@ -1,7 +1,7 @@
 prefix=@CMAKE_INSTALL_PREFIX@
-exec_prefix=${prefix}
-bindir=${prefix}/@RUNTIME_INSTALL_DIR@
-libdir=${prefix}/@LIBRARY_INSTALL_DIR@
+exec_prefix=@CMAKE_INSTALL_PREFIX@
```

Comment:
Thanks for having a look at this and submitting a PR. Looking at [this guide to pkg-config](https://people.freedesktop.org/~dbn/pkg-config-guide.html), there is nothing wrong of assigning `exec_prefix=${prefix}`. I would prefer if the CMake variable CMAKE_INSTALL_PREFIX would only be used once in this `package.pc.in` template file.

Do you maybe have a good reason for setting `exec_prefix` explicitly instead of to `${prefix}`?

Reply:
It's just more consistent with other packages. Here's some output from a pkgconfig directory:

```
mangix@mangix-pc:~/devstuff/openwrt/staging_dir/target-mips_24kc_musl/usr/lib/pkgconfig$ cat *| grep libdir=
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=/home/mangix/devstuff/openwrt/staging_dir/target-mips_24kc_musl/usr/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
typelibdir=${libdir}/girepository-1.0
libdir=${exec_prefix}/lib
typelibdir=${libdir}/girepository-1.0
libdir=${exec_prefix}/lib
typelibdir=${libdir}/girepository-1.0
libdir=${exec_prefix}/lib
typelibdir=${libdir}/girepository-1.0
libdir=${exec_prefix}/lib
typelibdir=${libdir}/girepository-1.0
libdir=${exec_prefix}/lib
typelibdir=${libdir}/girepository-1.0
libdir=${exec_prefix}/lib
typelibdir=${libdir}/girepository-1.0
libdir=${exec_prefix}/lib
typelibdir=${libdir}/girepository-1.0
libdir=${exec_prefix}/lib
typelibdir=${libdir}/girepository-1.0
libdir=${exec_prefix}/lib
typelibdir=${libdir}/girepository-1.0
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
typelibdir=${libdir}/girepository-1.0
libdir=${exec_prefix}/lib
typelibdir=${libdir}/girepository-1.0
libdir=${exec_prefix}/lib
typelibdir=${libdir}/girepository-1.0
libdir=${exec_prefix}/lib
typelibdir=${libdir}/girepository-1.0
libdir=${exec_prefix}/lib
typelibdir=${libdir}/girepository-1.0
libdir=${exec_prefix}/lib
typelibdir=${libdir}/girepository-1.0
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
pkglibdir=${libdir}/icu${ICULIBSUFFIX}/64.2
pkglibdir=${libdir}/icu${ICULIBSUFFIX}/64.2
pkglibdir=${libdir}/icu${ICULIBSUFFIX}/64.2
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
toolexeclibdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${prefix}/lib
libdir=${prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${prefix}/lib
libdir=${exec_prefix}/lib
libdir=${prefix}/lib/
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${prefix}/lib
libdir=${exec_prefix}/lib
libdir=${prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=/home/mangix/devstuff/openwrt/staging_dir/target-mips_24kc_musl/usr/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
xtlibdir=/usr/lib/iptables
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
libdir=${exec_prefix}/lib
sharedlibdir=${exec_prefix}/lib
```

Reply:
Yes, I can see why you propose to substitute `${prefix}` by `${exec_prefix}` and I am happy to accept this change. My comment relates only to the assignment of `exec_prefix=@CMAKE_INSTALL_PREFIX@` instead of `exec_prefix=${prefix}`. In other documentations such as the one I linked above, `exec_prefix=${prefix}` is common.

Would you mind reverting only this one line: `exec_prefix=...`? The other changes are fine.

Thanks!

Reply:
Sure I can do that. The predominant format is to specify it directly:

```
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=${prefix}
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=${prefix}
exec_prefix=${prefix}
exec_prefix=${prefix}
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=${prefix}
exec_prefix=/usr
exec_prefix=${prefix}
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr/bin
exec_prefix=/usr/bin
exec_prefix=${prefix}
exec_prefix=${prefix}
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=${prefix}
exec_prefix=/usr
exec_prefix=${prefix}
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=${prefix}
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=${prefix}
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=${prefix}
exec_prefix=/usr
exec_prefix=${prefix}
exec_prefix=${prefix}
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=${prefix}
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/home/mangix/devstuff/openwrt/staging_dir/target-mips_24kc_musl/usr
exec_prefix=${prefix}
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=${prefix}
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=/usr
exec_prefix=${prefix}
exec_prefix=/usr
exec_prefix=/usr
```